theano/tensor/basic.py

```diff
@@ -4181,12 +4186,11 @@ def infer_shape(self, node, ishapes):
         former case, the axis is fixed at construction, while in the
         latter it may vary over time depending on the value of the
         `axis` variable.
```

Comment:
Add a note section, that this function won't add the op in the graph when it is not useful.